packages/block-editor/src/components/off-canvas-editor/appender.js

```diff
@@ -41,6 +43,7 @@ export const Appender = forwardRef( ( props, ref ) => {
 				selectBlockOnInsert={ false }
 				shouldDirectInsert={ false }
 				__experimentalIsQuick
+				toggleLabel={ __( 'Add block to Navigation menu' ) }
```

Comment:
My concern with this is that it's another prop to pass to List View which will start to get unwieldy. Can we instead infer the block name from the context, and display it if we know it?

Reply:
I pushed a commit to do this. It's dirty, but it shows what's possible. The problem with this approach is that it modifies the title for all appenders. However that might be useful?

Reply:
Yeah, I'd prefer the dynamic approach, even if it changes the text everywhere. It might be worth getting a designer opinion on it though.

I also worked on #47047, which adds an aria-description to the appender button. That could also detail the parent block (rather than the label).

Reply:
Yes maybe that description approach would be best?

Reply:
I've gone with @talldan's suggestion.

I recommend we follow up with another PR to change the button text only _if deemed necessary_ by further a11y review.